Source/WindowsShellExtension/dllmain.cpp

```diff
@@ -0,0 +1,293 @@
+constexpr const wchar_t* exe_filename = L"MediaInfo.exe";
+
+BOOL APIENTRY DllMain(_In_ HMODULE hModule, _In_ DWORD  ul_reason_for_call, _In_opt_ LPVOID lpReserved) {
+    UNREFERENCED_PARAMETER(hModule);
+    UNREFERENCED_PARAMETER(lpReserved);
+    switch (ul_reason_for_call) {
+    case DLL_PROCESS_ATTACH:
+    case DLL_THREAD_ATTACH:
+    case DLL_THREAD_DETACH:
+    case DLL_PROCESS_DETACH:
+        break;
+    }
+    return TRUE;
+}
+
+namespace {
+    // Extracted from
+    // https://source.chromium.org/chromium/chromium/src/+/main:base/command_line.cc;l=109-159
+
+    std::wstring QuoteForCommandLineArg(_In_ const std::wstring& arg) {
+        // We follow the quoting rules of CommandLineToArgvW.
+        // http://msdn.microsoft.com/en-us/library/17w5ykft.aspx
+        std::wstring quotable_chars(L" \\\"");
+        if (arg.find_first_of(quotable_chars) == std::wstring::npos) {
+            // No quoting necessary.
+            return arg;
+        }
+
+        std::wstring out;
+        out.push_back('"');
+        for (size_t i = 0; i < arg.size(); ++i) {
+            if (arg[i] == '\\') {
+                // Find the extent of this run of backslashes.
+                size_t start = i, end = start + 1;
+                for (; end < arg.size() && arg[end] == '\\'; ++end) {}
+                size_t backslash_count = end - start;
+
+                // Backslashes are escapes only if the run is followed by a double quote.
+                // Since we also will end the string with a double quote, we escape for
+                // either a double quote or the end of the string.
+                if (end == arg.size() || arg[end] == '"') {
+                    // To quote, we need to output 2x as many backslashes.
+                    backslash_count *= 2;
+                }
+                for (size_t j = 0; j < backslash_count; ++j)
+                    out.push_back('\\');
+
+                // Advance i to one before the end to balance i++ in loop.
+                i = end - 1;
+            }
+            else if (arg[i] == '"') {
+                out.push_back('\\');
+                out.push_back('"');
+            }
+            else {
+                out.push_back(arg[i]);
+            }
+        }
+        out.push_back('"');
+
+        return out;
+    }
+
+    // Extracted from
+    // https://learn.microsoft.com/en-us/archive/msdn-magazine/2017/may/c-use-modern-c-to-access-the-windows-registry#reading-a-dword-value-from-the-registry
+    DWORD RegGetDword(HKEY hKey, const std::wstring& subKey, const std::wstring& value) {
+        DWORD data{};
+        DWORD dataSize = sizeof(data);
+        LONG retCode = RegGetValue(
+            hKey,
+            subKey.c_str(),
+            value.c_str(),
+            RRF_RT_REG_DWORD,
+            nullptr,
+            &data,
+            &dataSize
+        );
+        if (retCode != ERROR_SUCCESS)
+            throw std::runtime_error("Cannot read DWORD from registry.");
```

Comment:
I prefer silently quitting.

Reply:
The exception is caught at the place which calls this function. It is silently handled there.